test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTest.cs

```diff
@@ -0,0 +1,195 @@
+            Assert.Equal(1, configuredSources.Count);
+            Assert.Equal(configuration.Namespaces.Keys.First(), configuredSources[0]);
+        }
+
+        [Theory]
+        [InlineData("public, nuget", "nuge")]
+        [InlineData("public, nuget", "nuget1")]
+        [InlineData("public, Contoso.Opensource.*", "Cont")]
+        [InlineData("public, Contoso.Opensource.*", "Contoso.Opensource")]
+        public void SearchTree_WithOneSource_NoMatch(string packageNamespaces, string term)
+        {
+            // Arrange
+            SearchTree searchTree = GetSearchTree(packageNamespaces);
+
+            // Act & Assert
+            IReadOnlyList<string> configuredSources = searchTree.GetConfiguredPackageSources(term);
+            Assert.Null(configuredSources);
+        }
+
+        [Theory]
+        [InlineData("nuget.org, nuget | privateRepository, private*", "nuget")]
+        [InlineData("nuget.org, nuget |  privateRepository, private*", "private.")]
+        public void SearchTree_WithMultipleSources_Match(string packageNamespaces, string term)
+        {
+            // Arrange
+            PackageNamespacesConfiguration configuration = GetPackageNamespacesConfiguration(packageNamespaces);
+            SearchTree searchTree = new SearchTree(configuration);
+
+            // Act & Assert
+            IReadOnlyList<string> configuredSources = searchTree.GetConfiguredPackageSources(term);
+            Assert.Equal(1, configuredSources.Count);
+            Assert.True(configuredSources[0].StartsWith(term.Substring(0, 5)));
+        }
+
+        [Theory]
+        [InlineData("nuget.org,nuget | privateRepository,private*", "nuge")]
+        [InlineData("nuget.org,nuget | privateRepository,private*", "nuget1")]
+        [InlineData("nuget.org,nuget |  privateRepository,private*", "privat")]
+        public void SearchTree_WithMultipleSources_NoMatch(string packageNamespaces, string term)
+        {
+            // Arrange
+            SearchTree searchTree = GetSearchTree(packageNamespaces);
+
+            // Act & Assert
+            IReadOnlyList<string> configuredSources = searchTree.GetConfiguredPackageSources(term);
+            Assert.Null(configuredSources);
+        }
+
+        [Theory]
+        [InlineData("", "nuget")]
+        public void SearchTree_NoSources_NoMatch(string packageNamespaces, string term)
+        {
+            // Arrange
+            SearchTree searchTree = GetSearchTree(packageNamespaces);
+
+            // Act & Assert
+            IReadOnlyList<string> configuredSources = searchTree.GetConfiguredPackageSources(term);
+            Assert.Null(configuredSources);
+        }
+
+        [Fact]
+        public void SearchTree_TopNodeIsGlobbing_Match()
+        {
+            // Arrange
+            SearchTree searchTree = GetSearchTree("source1, nuget.* |  source2, nuget.common,nuget.protocol.* | source3, nuget.common.identity");
```

Comment:
This one is designed to test quite complex scenario so I didn't use [Theory] here.

<packageNamespaces>
    <packageSource key=""source1"">
        <namespace id=""nuget.*"" />
    </packageSource>
    <packageSource key=""source2"">
        <namespace id=""nuget.common"" />
        <namespace id=""nuget.protocol.*"" />
    </packageSource>
    <packageSource key=""source3"">
        <namespace id=""nuget.common.identity"" />
    </packageSource>
</packageNamespaces>